components/markdown_image_expand/markdown_image_expand.tsx

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {useState, useEffect} from 'react';
+import {CSSTransition} from 'react-transition-group';
+import './markdown_image_expand.scss';
+
+export type Props = {
+    alt: string;
+    children: React.ReactNode;
+    isEmbedVisible: boolean;
+    postId: string;
+};
+
+type State = {
+    isExpanded: boolean;
+};
+
+const MarkdownImageExpand: React.FC<Props> = ({children, alt, isEmbedVisible}: Props) => {
+    const [state, setState] = useState<State>({
+        isExpanded: isEmbedVisible,
+    });
+    const {isExpanded} = state;
+
+    useEffect(() => {
+        setState({
+            isExpanded: isEmbedVisible,
+        });
+    }, [setState, isEmbedVisible]);
+
+    const handleToggleButtonClick = () => {
+        setState((oldState) => ({...oldState, isExpanded: !oldState.isExpanded}));
+    };
+
+    const wrapperClassName = `markdown-image-expand ${isExpanded ? 'markdown-image-expand--expanded' : ''}`;
+
+    return (
+        <div className={wrapperClassName}>
+            <CSSTransition
+                in={isExpanded}
+                timeout={200}
+                mountOnEnter={true}
+                unmountOnExit={true}
+                classNames='markdown-image-expand__expanded-view'
+            >
+                <div>
```

Comment:
Is this div necessary or would it be possible to use a Fragment here to avoid the extra DOM node?

Reply:
Fixed.